test/jdk/javax/swing/JRootPane/DefaultButtonTest.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.lang.reflect.InvocationTargetException;
+
+/*
+ * @test
+ * @key headful
+ * @bug 8280913
+ * @summary Check whether the default button is honoured when <Enter> key is
+ * pressed when the focus is on the frame.
+ * @run main DefaultButtonTest
+ */
+public class DefaultButtonTest {
+    static JFrame frame = new JFrame();
+    static volatile boolean buttonPressed = false;
+    boolean testFailed = false;
+    JButton button1;
+    Robot robot;
+    private JPanel panel;
+    private JButton button2;
+
+    public static void main(String[] s) throws Exception {
+        DefaultButtonTest test = new DefaultButtonTest();
+        try {
+            test.runTest();
+        } finally {
+            if (frame != null) {
+                frame.dispose();
+                frame = null;
+            }
+        }
+
+    }
+
+    private static void setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+        } catch (UnsupportedLookAndFeelException ignored) {
+            System.out.println("Unsupported L&F: " + laf.getClassName());
+        } catch (ClassNotFoundException | InstantiationException
+                | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private void createUI() {
+        panel = new JPanel();
+        panel.setLayout(new FlowLayout());
+        button1 = new JButton("butt1");
+        button1.addActionListener(new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                synchronized (this) {
+                    buttonPressed = true;
+                    notifyAll();
+                }
+            }
+        });
+        panel.add(button1);
+
+        button2 = new JButton("butt2");
+        panel.add(button2);
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        frame.setLayout(new FlowLayout());
+
+        frame.add(panel);
+
+        frame.setSize(200, 300);
+        frame.setLocationRelativeTo(null);
+        frame.getRootPane().setDefaultButton(button1);
+        frame.setVisible(true);
+    }
+
+    public void runTest() throws InvocationTargetException, InterruptedException {
+        try {
+            robot = new Robot();
+        } catch (Exception e) {
+            System.err.print("Error creating robot");
+            e.printStackTrace();
+            System.exit(1);
```

Comment:
You must never call `System.exit` from a jtreg test: just throw the exception.

So in this case, you don't need try-catch here, just let `AWTException` propagate to main.